service-tests/shippable.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const isBuildStatus = Joi.string().regex(/^(passing|failing|cancelled|unstable|pending|skipped|none)$/);
```

Comment:
This could be a good addition to [`validators.js`](https://github.com/badges/shields/blob/master/service-tests/helpers/validators.js) as i'm sure that other badges would also benifit from this.

Reply:
Unfortunately, all the CI integrations seem to define a subtly different selection of labels/terminology. e.g:

https://github.com/badges/shields/blob/master/server.js#L462-L474
https://github.com/badges/shields/blob/master/server.js#L4761-L4777
https://github.com/badges/shields/blob/master/server.js#L4999-L5019

Reply:
Is **none** a valid build status? Seems to me it's not. Could we remove it from the regex?

Reply:
Looks sensible to me:

https://github.com/badges/shields/blob/9cb9d5984d38929662cecf8b0c890c83767ea35b/server.js#L535-L536

https://github.com/badges/shields/blob/9cb9d5984d38929662cecf8b0c890c83767ea35b/server.js#L572-L575

Sorry - the links I posted above in response to @RedSparr0w 's comment are now confusing because I forgot to pin them to a commit and the line numbers have now changed.

Reply:
I dug through their API docs for a while, and couldn't find the statuses listed anywhere, though I did find the dead link in archive.org. I sent them a tweet to see if there's a replacement.

- http://web.archive.org/web/20170627205023/http://docs.shippable.com/ci/build_status/
- http://docs.shippable.com/platform/api/api-overview/?#!/Projects/get_projects_projectId_branchRunStatus

Should we switch to that status API?

Reply:
I've had a look through the API docs. I'm not sure this makes anything clearer :(

The statuses in the existing SVG parsing code we've got a case for 'skipped' which doesn't seem to map on to any of the codes listed on http://web.archive.org/web/20170627205023/http://docs.shippable.com/ci/build_status/

Conversely, the 'timeout' status listed on http://web.archive.org/web/20170627205023/http://docs.shippable.com/ci/build_status/ doesn't seem to map on to any of the statuses we currently handle in the SVG parsing code.

'none' does seem to be a label that can be returned by the badge endpoint in some cases e.g: https://api.shippable.com/projects/5701519833e2f1203f8ca8d0/badge

It is odd that the API status codes aren't clearly documented anywhere. It appears to be an issue that others have raised: https://github.com/Shippable/support/issues/2560 (links are dead now). Let me know if they get back to you to clarify.

Switching to the API probably means slightly changing the statuses the badge can possibly report. If we can get clarity on the codes, maybe it is better to just make that change?

Reply:
I sent them another tweet… hopefully will get a response!

Reply:
@paulmelnikow My apologies for the missing documentation. This got left out somehow when we restructured our docs. I'll get that updated very soon, but in the mean time I hope this helps:

```
0:  "WAITING"
10: "QUEUED"
20: "PROCESSING"
30: "SUCCESS"
40: "SKIPPED"
50: "UNSTABLE"
60: "TIMEOUT"
70: "CANCELED"
80: "FAILED"
90: "STOPPED"
```

Reply:
Thanks @rageshkrishna - that's useful :)

@paulmelnikow I'll aim to pick this up tomorrow evening